upgrade_test.py

```diff
@@ -31,50 +31,105 @@ class UpgradeTest(FillDatabaseData):
             # backup the data
             node.remoter.run('sudo cp /etc/scylla/scylla.yaml /etc/scylla/scylla.yaml-backup')
+            # flush all memtables to SSTables
+            node.remoter.run('sudo nodetool drain')
             node.remoter.run('sudo nodetool snapshot')
+            node.remoter.run('sudo systemctl stop scylla-server.service')
             node.remoter.run('sudo chown root.root /etc/yum.repos.d/scylla.repo')
             node.remoter.run('sudo chmod 644 /etc/yum.repos.d/scylla.repo')
             node.remoter.run('sudo yum clean all')
             ver_suffix = '-{}'.format(new_version) if new_version else ''
-            node.remoter.run('sudo yum install scylla{0} scylla-server{0} scylla-jmx{0} scylla-tools{0}'
-                             ' scylla-conf{0} scylla-kernel-conf{0} scylla-debuginfo{0} -y'.format(ver_suffix))
+            if upgrade_mode == 'direct':
+                node.remoter.run('sudo yum remove scylla\* -y'.format(ver_suffix))
+                node.remoter.run('sudo yum install scylla{0} -y'.format(ver_suffix))
+            else:
+                node.remoter.run('sudo yum update scylla{0}\* -y'.format(ver_suffix))
+        node.remoter.run('sudo systemctl start scylla-server.service')
+        node.wait_db_up(verbose=True)
+        result = node.remoter.run('rpm -qa scylla\*server')
+        new_ver = result.stdout
+        assert self.orig_ver != self.new_ver, "scylla-server version isn't changed"
+        self.new_ver = new_ver
+
+    def rollback_node(self, node):
+        self.log.info('Rollbacking a Node')
+        upgrade_mode = self.db_cluster.params.get('upgrade_mode', default='smooth')
+        new_introduced_pkgs = self.db_cluster.params.get('new_introduced_pkgs', default='scylla\*-tools-core')
+        result = node.remoter.run('scylla --version')
+        orig_ver = result.stdout
+        node.remoter.run('sudo cp ~/scylla.repo-backup /etc/yum.repos.d/scylla.repo')
         # flush all memtables to SSTables
-        node.remoter.run('sudo nodetool drain')
-        node.remoter.run('sudo systemctl restart scylla-server.service')
+        node.remoter.run('nodetool drain')
+        # backup the data
+        node.remoter.run('nodetool snapshot')
+        node.remoter.run('sudo systemctl stop scylla-server.service')
+        node.remoter.run('sudo chown root.root /etc/yum.repos.d/scylla.repo')
+        node.remoter.run('sudo chmod 644 /etc/yum.repos.d/scylla.repo')
+        node.remoter.run('sudo yum clean all')
+        if upgrade_mode == 'direct':
+            node.remoter.run('sudo yum remove scylla\* -y')
+            node.remoter.run('sudo yum install {} -y' % node.scylla_pkg())
+        elif upgrade_mode == 'maintenance':
+            node.remoter.run('sudo yum downgrade scylla\*%s -y' % self.orig_ver.split('-')[0])
+        else:
+            if new_introduced_pkgs:
+                node.remoter.run('sudo yum remove %s -y' % new_introduced_pkgs)
```

Comment:
Can't we just remove everything that is coming from scylla?
Without this being a parameter...